### ISSUE TITLE
Failed to parse tags for entity

### ISSUE THREAD

**Issue by chrisjaimon2012** [bridgecrewio/checkov#1533]
**Describe the bug**
Parser for CloudFormation throws a warning when tag keys have intrinsic functions
```yaml
Tags:
  - Key: testkey
    Value: !Sub "${testparameter}-value"
  - Key: !Sub "${testparameter}-key"
    Value: testvalue
```
Here tag1 where Key: testkey works properly, but the second tag throws the warning
```bash
[MainThread  ] [WARNI]  Failed to parse tags for entity
```

**To Reproduce**
1. Use the below cloudformation template
```yaml
AWSTemplateFormatVersion: 2010-09-09

Parameters:
  testparameter:
    Type: String
    Default: test

Resources:
  S3Bucket:
    Type: 'AWS::S3::Bucket'
    Properties:
      Tags:
        - Key: testkey
          Value: !Sub "${testparameter}-value"
        - Key: !Sub "${testparameter}-key"
          Value: testvalue
```
2. Run cli command  `checkov -f filename.yaml`
3. Error
```
2021-08-25 18:53:48,589 [MainThread  ] [WARNI]  Failed to parse tags for entity {'S3Bucket': {'Type': 'AWS::S3::Bucket', 'Properties': {'Tags': [{'Key': 'testkey', 'Value': {'Fn::Sub': '${testparameter}-value'}, '__startline__': 13, '__endline__': 15}, {'Key': {'Fn::Sub': '${testparameter}-key'}, 'Value': 'testvalue', '__startline__': 15, '__endline__': 17}], '__startline__': 10, '__endline__': 17, 'resource_type': 'AWS::S3::Bucket'}}}
```

**Expected behavior**
Tag keys using CF intrinsic functions should not throw a warning

**Additional context**
https://github.com/bridgecrewio/checkov/blob/2f0c5cc888078f3a2886c65771d4b9a7c7f2ceeb/checkov/cloudformation/cfn_utils.py#L45
i guess 👆🏻  should be 👇🏻 
```python
tag_dict = {get_entity_value_as_string(tag["Key"]): get_entity_value_as_string(tag["Value"]) for tag in tags}
```
